scheduler/reconciler/reconcile_node.go

```diff
@@ -441,7 +441,15 @@ func (nr *NodeReconciler) computeForNode(
 			continue
 		}
 
-		nr.createDeployment(job, tg, dstate, len(result.Update), liveAllocs)
+		maxParallel := 1
+		if !tg.Update.IsEmpty() {
+			maxParallel = tg.Update.MaxParallel
+		}
+
+		// maxParallel of 0 means no deployments
+		if maxParallel != 0 {
+			nr.createDeployment(job, tg, dstate, len(result.Update), liveAllocs)
```

Comment:
If I'm looking at this later, I'm going to wonder why if this is conditional on `maxParallel != 0`, doesn't the `isDeploymentComplete` method need to be as well. So maybe we should leave a comment here pointing out that even if there's no deployment we mark the deployment complete?